src/Symfony/Component/HttpKernel/DataCollector/RequestDataCollector.php

```diff
@@ -495,4 +497,76 @@ private function parseController(array|object|string|null $controller): array|st
 
         return \is_string($controller) ? $controller : 'n/a';
     }
+
+    public function getCurlCommand(): ?string
+    {
+        if (!isset($this->data['method']) || !isset($this->data['path_info'])) {
+            return null;
+        }
+
+        $command = ['curl', '--compressed'];
+
+        // Build the full URL
+        $scheme = $this->data['request_server']['HTTPS'] ?? false ? 'https' : 'http';
+        $host = $this->data['request_server']['HTTP_HOST'] ?? $this->data['request_server']['SERVER_NAME'] ?? 'localhost';
```

Comment:
this is the wrong way to implement this, as it won't take trusted proxies into account.
The appropriate getters of `Request` must be used (which might imply collecting them in the data if not done already)